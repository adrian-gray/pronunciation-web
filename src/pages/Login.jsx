import React from "react";
import { Redirect } from "react-router-dom";
import { auth } from "./../firebase";

import Button from "@material-ui/core/Button";
import Divider from "@material-ui/core/Divider";
import Paper from "@material-ui/core/Paper";
import TextField from "@material-ui/core/TextField";
import Typography from "@material-ui/core/Typography";
import withStyles from "@material-ui/styles/withStyles";

const styles = theme => ({
  page: theme.page,
  headspace: theme.headspace,
  personalSpace: theme.personalSpace,
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit
  }
});

class Login extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      signupName: "",
      signupEmail: "",
      signinEmail: "",
      signupPassword: "",
      signinPassword: ""
    };

    this.handleChange = this.handleChange.bind(this);
    this.signIn = this.signIn.bind(this);
    this.signUp = this.signUp.bind(this);
  }

  handleChange(name) {
    return e => {
      this.setState({
        [name]: e.target.value
      });
    };
  }

  signUp() {
    auth.signUpWithEmail({
      name: this.state.signupName,
      email: this.state.signupEmail,
      password: this.state.signupPassword
    });
  }

  signIn() {
    auth.signInWithEmail({
      email: this.state.signinEmail,
      password: this.state.signinPassword
    });
  }

  render() {
    const { classes, user } = this.props;
    if (user) {
      return <Redirect to="/home" />;
    }

    return (
      <Paper className={classes.page}>
        <div>
          <Paper className={classes.personalSpace}>
            <Button
              onClick={auth.handleGoogleLogin}
              variant="contained"
              color="primary"
              className={classes.button}
            >
              {`Login with Google`}
            </Button>
          </Paper>
        </div>
        <div>
          <Paper className={classes.personalSpace}>
            <Typography className={classes.headSpace} gutterBottom>
              {`Sign in with email`}
            </Typography>
            <form>
              <TextField
                id="signin_email"
                label="Your email"
                className="form-control"
                value={this.state.signinEmail}
                onChange={this.handleChange("signinEmail")}
                margin="normal"
              />
              <TextField
                id="signin_password"
                label="Password"
                className="form-control"
                type="password"
                value={this.state.signinPassword}
                onChange={this.handleChange("signinPassword")}
                margin="normal"
              />
              <Button onClick={this.signIn} variant="raised">
                {"Sign in"}
              </Button>
            </form>
          </Paper>
          <Divider />
        </div>
        <Paper className={classes.personalSpace}>
          <Typography className={classes.headspace} gutterBottom>
            {`Or, sign up with email `}
          </Typography>
          <form>
            <TextField
              id="signup_name"
              label="Your name"
              className="form-control"
              value={this.state.signupName}
              onChange={this.handleChange("signupName")}
              margin="normal"
            />
            <TextField
              id="signup_email"
              label="Your email"
              className="form-control"
              value={this.state.signupEmail}
              onChange={this.handleChange("signupEmail")}
              margin="normal"
            />
            <TextField
              id="signup_password"
              label="Password"
              className="form-control"
              type="password"
              value={this.state.signupPassword}
              onChange={this.handleChange("signupPassword")}
              margin="normal"
            />
            <Button onClick={this.signUp} variant="raised">
              {"Sign up"}
            </Button>
          </form>
        </Paper>
      </Paper>
    );
  }
}

export default withStyles(styles)(Login);
