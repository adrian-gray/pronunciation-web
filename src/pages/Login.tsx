import React, { useState, useEffect } from "react";
import { Redirect } from "react-router-dom";
import { auth } from "./../firebase";

import { IUser } from "./../../@types/PronounceWeb";

import Button from "react-bootstrap/Button";
import Container from "react-bootstrap/Container";
import Form from "react-bootstrap/Form";

export default (props: { user: IUser }) => {
  const [signupName, setSignupName] = useState("");
  const [signupEmail, setSignupEmail] = useState("");
  const [signinEmail, setSigninEmail] = useState("");
  const [signupPassword, setSignupPassword] = useState("");
  const [signinPassword, setSigninPassword] = useState("");
  const [redirectToHome, setRedirectToHome] = useState(false);

  useEffect(() => {
    if (props.user) {
      setRedirectToHome(true);
    }
  });

  const handleChange = (property: string) => {
    return (e: React.FormEvent) => {
      const element = e.target as HTMLInputElement;
      const value = element.value;
      switch (property) {
        case "signupName":
          setSignupName(value);
          break;
        case "signupEmail":
          setSignupEmail(value);
          break;
        case "signinEmail":
          setSigninEmail(value);
          break;
        case "signupPassword":
          setSignupPassword(value);
          break;
        case "signinPassword":
          setSigninPassword(value);
          break;
        default:
          console.log(`Erm. what's a ${property}?`);
      }
    };
  };

  const signUp = () => {
    auth.signUpWithEmail({
      name: signupName,
      email: signupEmail,
      password: signupPassword
    });
  };

  const signIn = () => {
    auth.signInWithEmail({
      email: signinEmail,
      password: signinPassword
    });
  };

  return (
    <Container>
      {redirectToHome ? <Redirect to="/home" /> : null}
      <div>
        <Container className="personal-space">
          <Button variant="primary" onClick={auth.handleGoogleLogin}>
            {`Login with Google`}
          </Button>
        </Container>
      </div>
      <div>
        <Container className="personal-space">
          <p className="head-space">{`Sign in with email`}</p>
          <Form>
            <Form.Row>
              <Form.Label>Your email</Form.Label>
              <Form.Control
                id="signin_email"
                className="form-control"
                value={signinEmail}
                onChange={handleChange("signinEmail")}
                autoComplete="username"
              />
              <Form.Label>Password</Form.Label>
              <Form.Control
                id="signin_password"
                className="form-control"
                type="password"
                value={signinPassword}
                onChange={handleChange("signinPassword")}
                autoComplete="current-password"
              />
              <Button onClick={signIn}>{"Sign in"}</Button>
            </Form.Row>
          </Form>
        </Container>
        <hr />
      </div>
      <Container className="personal-space">
        <p className="head-space">{`Or, sign up with email `}</p>
        <Form>
          <Form.Row>
            <Form.Label>Your name</Form.Label>
            <Form.Control
              id="signup_name"
              className="form-control"
              value={signupName}
              onChange={handleChange("signupName")}
            />
          </Form.Row>
          <Form.Row>
            <Form.Label>Your email</Form.Label>
            <Form.Control
              id="signup_email"
              className="form-control"
              value={signupEmail}
              onChange={handleChange("signupEmail")}
              autoComplete="username"
            />
          </Form.Row>
          <Form.Row>
            <Form.Label>Password</Form.Label>
            <Form.Control
              autoComplete="current-password"
              id="signup_password"
              className="form-control"
              type="password"
              value={signupPassword}
              onChange={handleChange("signupPassword")}
            />
          </Form.Row>
          <Form.Row>
            <Button onClick={signUp}>{"Sign up"}</Button>
          </Form.Row>
        </Form>
      </Container>
    </Container>
  );
};
