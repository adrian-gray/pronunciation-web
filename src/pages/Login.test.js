import React from "react";
import { shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import Login from "./Login";

afterEach(cleanup);

test("Login", async () => {
  const props = {};

  const container = shallow(<Login {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
