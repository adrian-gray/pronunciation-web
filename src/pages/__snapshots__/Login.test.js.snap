// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login 1`] = `
<div>
  <div
    class="headspace"
  >
    <div
      class="personal-space"
    >
      <div>
        <div
          class="google-button"
        >
          <div
            class="google-signin-wrapper"
          >
            <div
              class="google-signin-button-icon"
            >
              <div
                class="google-signin-button-image abcRioButtonSvgImageWithFallback abcRioButtonIconImage abcRioButtonIconImage18"
              >
                <svg
                  class="google-svg"
                  height="18px"
                  version="1.1"
                  viewBox="0 0 48 48"
                  width="18px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M24 9.5c3.54 0 6.71 1.22 9.21 3.6l6.85-6.85C35.9 2.38 30.47 0 24 0 14.62 0 6.51 5.38 2.56 13.22l7.98 6.19C12.43 13.72 17.74 9.5 24 9.5z"
                      fill="#EA4335"
                    />
                    <path
                      d="M46.98 24.55c0-1.57-.15-3.09-.38-4.55H24v9.02h12.94c-.58 2.96-2.26 5.48-4.78 7.18l7.73 6c4.51-4.18 7.09-10.36 7.09-17.65z"
                      fill="#4285F4"
                    />
                    <path
                      d="M10.53 28.59c-.48-1.45-.76-2.99-.76-4.59s.27-3.14.76-4.59l-7.98-6.19C.92 16.46 0 20.12 0 24c0 3.88.92 7.54 2.56 10.78l7.97-6.19z"
                      fill="#FBBC05"
                    />
                    <path
                      d="M24 48c6.48 0 11.93-2.13 15.89-5.81l-7.73-6c-2.15 1.45-4.92 2.3-8.16 2.3-6.26 0-11.57-4.22-13.47-9.91l-7.98 6.19C6.51 42.62 14.62 48 24 48z"
                      fill="#34A853"
                    />
                    <path
                      d="M0 0h48v48H0z"
                      fill="none"
                    />
                  </g>
                </svg>
              </div>
            </div>
            <div
              class="google-button-contents"
            >
              Sign in with Google
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="personal-space"
      >
        <p
          class="head-space"
        >
          Sign in with email
        </p>
        <form
          class=""
        >
          <div
            class="form-row"
          >
            <label
              class="form-label"
            >
              Your email
            </label>
            <input
              autocomplete="username"
              class="form-control form-bottom-space form-control"
              id="signin_email"
              value=""
            />
            <label
              class="form-label"
            >
              Password
            </label>
            <input
              autocomplete="current-password"
              class="form-control form-bottom-space form-control"
              id="signin_password"
              type="password"
              value=""
            />
            <button
              class="btn btn-primary"
              type="button"
            >
              Sign in
            </button>
          </div>
        </form>
      </div>
      <hr />
    </div>
    <div
      class="personal-space"
    >
      <p
        class="head-space"
      >
        Or, sign up with email 
      </p>
      <form
        class=""
      >
        <div
          class="form-row"
        >
          <label
            class="form-label"
          >
            Your name
          </label>
          <input
            class="form-control form-bottom-space form-control"
            id="signup_name"
            value=""
          />
        </div>
        <div
          class="form-row"
        >
          <label
            class="form-label"
          >
            Your email
          </label>
          <input
            autocomplete="username"
            class="form-control form-bottom-space form-control"
            id="signup_email"
            value=""
          />
        </div>
        <div
          class="form-row"
        >
          <label
            class="form-label"
          >
            Password
          </label>
          <input
            autocomplete="current-password"
            class="form-control form-bottom-space form-control"
            id="signup_password"
            type="password"
            value=""
          />
        </div>
        <div
          class="form-row"
        >
          <button
            class="btn btn-primary"
            type="button"
          >
            Sign up
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
